File: app/models/address.rb
class Address < ApplicationRecord
  belongs_to :user, optional: true

  with_options presence: true do
    validates :postal_code, format: { with: /\A\d{7}\z/, message: "を正しく入力してください"}
    validates :prefecture
    validates :city
    validates :town_name
    validates :house_number
  end
end

File: app/javascript/address.js
$(function() {
  return $('#postal_code').jpostal({
    postcode: ['#postal_code'],
    address: {
      '#prefecture' : '%3',
      '#city' : '%4',
      '#town_name' : '%5',
      '#house_number' : '%6%7'
    },
  });
});